internal/controller/grafanaorganization_controller.go

```diff
@@ -104,11 +96,82 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 		return ctrl.Result{}, errors.WithStack(err)
 	}
 
-	//TODO Implement the logic to create the Grafana organization
+	// if the CR doesn't have an orgID, create the organization in Grafana and update the status
+	if grafanaOrganization.Status.OrgID == 0 {
+		return ctrl.Result{}, r.createOrganizationInGrafana(ctx, grafanaOrganization)
+	} else {
+		searchResult, err := r.GrafanaAPI.Orgs.GetOrgByID(grafanaOrganization.Status.OrgID)
+		if err != nil {
+			// Parsing error message to find out the error code
+			is404 := strings.Contains(err.Error(), "(status 404)")
```

Comment:
I don't really like that this error management is duplicated but I don't see a better way yet

Reply:
I don't like it neither :/

Reply:
I would do the id and name check first

Reply:
If the name is not available, wheter you create or update, then this should error

Reply:
Because if you update org1 into org2 but display name org2 is already taken, this needs to be tested

Reply:
So I refactored the code, putting the name check first, but I didn't find a way to avoid duplicating the 404 error check for both `getOrgByName` and `getOrgByID`